src/app/components/phrase-diff/phrase-diff.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { Observable } from "rxjs";
+import { Edit, ArticleComparisonService } from "../../services/article-comparison.service";
+
+export interface PhraseDiffProps {
+    edits: Edit[];
+}
+
+@Component({
+    selector: "app-phrase-diff",
+    templateUrl: "./phrase-diff.component.html",
+    styleUrls: ["./phrase-diff.component.sass"],
+})
+export class PhraseDiffComponent implements OnInit {
+    @Input() recordedPhrase: Observable<string>;
+    @Input() groundTruth: string;
+
+    edits: Edit[] = [];
+
+    constructor(public diffService: ArticleComparisonService) {}
+
+    ngOnInit(): void {
```

Comment:
From my understanding, I just want to subscribe to the Observable once, so putting this subscription in ngOnInit makes sense since it only subscribes once after the component is initialised with its input properties.

(Also I should have unsubscribed somewhere as well to better handle memory resources...)

But given that this component will no longer be receiving an Observable, the lifecycle methods won't be needed.

Reply:
Re the memory part, you don't need to unsubscribe from the observable if it is a HttpClient method calls since it limits the RxJS Observable to execute once.